sdk/profiles/src/main/java/io/opentelemetry/sdk/profiles/data/ProfileData.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.profiles.data;
+
+import io.opentelemetry.api.common.Attributes;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Represents a complete profile, including sample types, samples, mappings to binaries, locations,
+ * functions, string table, and additional metadata.
+ *
+ * @see "pprofextended.proto::Profile"
+ */
+@Immutable
+public interface ProfileData {
```

Comment:
That's going to make the marshallers, which are the primary users of this API, much more complex. They only care about the indexes, not the values. If all they can get is the value, they have to do reverse lookup on the table, which they don't have. So now they all need to couple to the string table too?
The current API is horrible for users who want Strings, but they are not its intended audience. It's fit for its intended use as is, the proposed change makes it less so. It's viable to add alternative getters for the String without removing the index ones, but see notes elsewhere on issues implementing those.

Reply:
Assuming you're ok living without this, now that the revised plan is to move these classes into the serialization package for use just by marshallers, rather than having them for general use in the sdk package.